packages/cpp/src/browser/cpp-commands.ts

```diff
@@ -40,8 +41,13 @@ export class CppCommandContribution implements CommandContribution {
 
     registerCommands(commands: CommandRegistry): void {
         commands.registerCommand(SWITCH_SOURCE_HEADER, {
-            isEnabled: () => (this.editorService && !!this.editorService.activeEditor &&
-                (this.editorService.activeEditor.editor.document.uri.endsWith(".cpp") || this.editorService.activeEditor.editor.document.uri.endsWith(".h"))),
+            isEnabled: () => {
+                if (this.editorService && !!this.editorService.activeEditor) {
```

Comment:
This check is duplicated, here and in cpp-keybindings.ts, so I would suggest moving it to a function `activeEditorIsCppFile(editorManager: EditorManager)` or something like that.